### ISSUE TITLE
Confusing error

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#2554]
I am not sure whether it's an issue with pyright or pylance, so I decided to report here.

Code:
```py
from typing import overload

@overload
def f(x: int, y: str) -> None:
    ...

@overload
def f(x: str, y: int) -> None:
    ...

def f(x, y) -> None:
    print(x, y, y, x)

f(x=5, "4")
```

If you hover over `f`, you get this error:
![image](https://user-images.githubusercontent.com/42166884/141176468-aaa6c569-f702-482d-ad33-571533b08c5b.png)

If you hover over `x`:
![image](https://user-images.githubusercontent.com/42166884/141176507-5c1bf385-28c8-4eec-b5f3-b73839073f99.png)

If you hover over the last argument you see the real issue:
![image](https://user-images.githubusercontent.com/42166884/141176408-97f4f932-1538-4acb-aa98-fb2fd7431a7a.png)

How does pyright even know what arguments I'm passing if the call is illegal?
If one didn't know this was illegal in Python one would probably think this call refers to the topmost overload. And this is what the second error is suggesting. So I think this is pretty confusing.

Pyright also reports two issues in the CLI:
```
  /home/.../foo.py:16:8 - error: Positional argument cannot appear after keyword arguments
  /home/.../foo.py:16:1 - error: No overloads for "f" match the provided arguments
  ```

**Comment by erictraut** [comment]
The issue here is that one of the errors is coming from the parser, which is detecting a syntax error. Unlike the Python interpreter, pyright doesn't simply stop when it detects a syntax error. It builds the best parse tree it's able to generate and the runs semantic analysis on that parse tree. This is important for a language server because syntax errors occur frequently while you're in the middle of typing a line of code, yet the language server needs to perform semantic analysis so it can offer valid completion suggestions and signature help. 

So that explains why you're seeing two errors here. This is expected behavior.
